### ISSUE TITLE
dependency on kustomize causing compilation issues

### ISSUE THREAD

**Issue by pvenu-Google** [kubernetes/cli-runtime#19]
Relevant section of the go.mod file is as below -:
        k8s.io/api v0.19.4
        k8s.io/apiextensions-apiserver v0.19.4 // indirect
        k8s.io/apimachinery v0.19.4
        k8s.io/cli-runtime v0.19.4
        k8s.io/client-go v0.19.4

On compiling some kubectl plugin code, i get this error -:
go vet ./...
# sigs.k8s.io/kustomize/pkg/transformers/config
vendor/sigs.k8s.io/kustomize/pkg/transformers/config/factorycrd.go:71:47: cannot use api.Schema.SchemaProps.Properties (type spec.SchemaProperties) as type myProperties in argument to looksLikeAk8sType
make: *** [Makefile:95: vet] Error 2

Graph dependency shows -:
go mod graph | grep kustomize
k8s.io/cli-runtime@v0.19.4 sigs.k8s.io/kustomize@v2.0.3+incompatible

Similar issue also noted by helm package users
https://github.com/kubernetes-sigs/kustomize/issues/3262

Looks like the latest version v0.21.0 uses a fixed version of kustomize in the go.mod file. 
Could we get a patch release for v0.19.x or v0.20.x ?
Its going to be a while till we get to use kubernetes1.21 and it seems that anything below cli-runtime@v0.21.0 is unusable at this time ? If not, please let me know what i could do as a work around ? 

The kustomize version in the mod file is quite old and it looks like the kustomize package structure has changed along the way.

**Comment by monopole** [comment]
i wrote up some notes here:
https://github.com/kubernetes-sigs/kustomize/tree/master/cmd/depprobcheck

**Comment by monopole** [comment]
tl;dr anyone depending on `k8s.io` _v0.20.x_ (or older) packages will have to avoid depending on anything that depends on  `k8s.io/kube-openapi` past _v0.0.0-20210323165736-1a6458611d18_.

**Comment by pvenu-Google** [comment]
Thanks monopole. 
So i am using cli-runtime 19.4 which seems to be using a much older version of kube-openapi, but i still see the issue. 

go mod graph | grep kube-openapi
k8s.io/code-generator@v0.19.2 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/apimachinery@v0.19.4 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/apiserver@v0.19.4 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/apiextensions-apiserver@v0.19.4 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/cli-runtime@v0.19.4 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/code-generator@v0.19.4 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/apiserver@v0.19.2 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/apiextensions-apiserver@v0.19.2 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 sigs.k8s.io/yaml@v1.1.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 sigs.k8s.io/structured-merge-diff/v4@v4.0.1
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 k8s.io/gengo@v0.0.0-20200413195148-3a45101e95ac
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 gopkg.in/yaml.v2@v2.2.2
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 golang.org/x/tools@v0.0.0-20181011042414-1f849cf54d09
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 golang.org/x/net@v0.0.0-20200324143707-d3edc9973b7e
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/stretchr/testify@v1.3.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/spf13/pflag@v0.0.0-20170130214245-9ff6c6923cff
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/onsi/gomega@v0.0.0-20170829124025-dcabb60a477c
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/onsi/ginkgo@v0.0.0-20170829012221-11459a886d9c
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/munnerz/goautoneg@v0.0.0-20120707110453-a547fc61f48d
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/mailru/easyjson@v0.0.0-20160728113105-d5b7844b561a
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/json-iterator/go@v1.1.6
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/googleapis/gnostic@v0.4.1
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/google/gofuzz@v1.0.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/google/go-cmp@v0.4.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/golang/protobuf@v1.3.3
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/go-openapi/swag@v0.0.0-20160704191624-1d0bd113de87
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/go-openapi/spec@v0.0.0-20160808142527-6aced65f8501
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/go-openapi/jsonreference@v0.0.0-20160704190145-13c6e3589ad9
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/go-openapi/jsonpointer@v0.0.0-20160704185906-46af16f9f7b1
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/ghodss/yaml@v0.0.0-20150909031657-73d445a93680
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/emicklei/go-restful@v0.0.0-20170410110728-ff4f55a20633
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/davecgh/go-spew@v1.1.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/PuerkitoBio/urlesc@v0.0.0-20160726150825-5bd2802263f2
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/PuerkitoBio/purell@v1.0.0
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 github.com/NYTimes/gziphandler@v0.0.0-20170623195520-56545f4a5d46
k8s.io/apimachinery@v0.19.2 k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6
k8s.io/kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 k8s.io/klog/v2@v2.0.0

**Comment by pvenu-Google** [comment]
The reason for kube-openapi@v0.0.0-20200805222855-6aeccd4b50c6 not working even though it was older than the recommended v0.0.0-20210323165736-1a6458611d18 was because the go-openapi/spec package also had a version dependency. 

This commit changed the properties field type in the schema struct which caused the same error to pop up.

https://github.com/go-openapi/spec/commit/55f43acfece4ec21dd910b355e80e15d35960aa9

I used v0.19.6 and the compilation issue went away

**Comment by monopole** [comment]
IMO, this can be closed.

**Comment by pvenu-Google** [comment]
yes, please do. Thanks.
